flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/itcases/AbstractQueryableStateTestBase.java

```diff
@@ -1307,24 +1268,13 @@ JobID getJobId() {
 		public void close() throws Exception {
 			// Free cluster resources
 			if (jobId != null) {
-				cluster.getLeaderGateway(deadline.timeLeft())
-						.ask(new JobManagerMessages.CancelJob(jobId), deadline.timeLeft())
-						.mapTo(ClassTag$.MODULE$.<CancellationSuccess>apply(CancellationSuccess.class));
-
-				cancellationFuture.get(deadline.timeLeft().toMillis(), TimeUnit.MILLISECONDS);
+				clusterClient.cancel(jobId);
+				// cancel() is non-blocking so do this to make sure the job finished
+				clusterClient.requestJobResult(jobId).get();
```

Comment:
add timeout. We should also check the returned status to make sure it is actually finished.

Reply:
unfortunately, we can't determine whether the job finished from that. But the result is only available of the job finished somehow, either naturally, or because of failure or because of cancel.

Reply:
couldn't we query the job status instead?